devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,249 @@
+/*
+ *  Copyright 2017 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  ZigBee Ceiling Fan Light
+ *
+ *  Author: SAMSUNG LED
+ *  Date: 2021-06-30
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+import groovy.json.JsonOutput
+
+metadata {
+	definition (name: "LED FAN lightings", namespace: "SAMSUNG LED", author: "SAMSUNG LED", runLocally: true, minHubCoreVersion: '000.019.00012', executeCommandsLocally: true, genericHandler: "Zigbee") {
+
+		capability "Actuator"		
+		capability "Configuration"
+		capability "Health Check"
+		capability "Refresh"
+		capability "Switch"
+		capability "Switch Level"
```

Comment:
Why do use this capability as a "bridge" to a Fan Speed, when you can just use `capability "Fan Speed"`? 
Using 0-100 range to control 5 speeds doesn't seem to be a great UX approach.

Reply:
Because we want connect other platforms via SmartThings cloud to cloud connection. But other platforms don't support "Fan speed" capability. So we have to use "Switch level" capability for control fan speed.

Reply:
I don't like it, but if it have to be this way, so be it. Would it be a problem for this "other platform" if you redefined range of `Switch Level` capability to be 0-4, instead of 0-100?

Reply:
I'm fully agree with you. I don't like it too. Other platforms only support switch level range 0 to 100. I can't change the level range. It is absurd but there is no way.